### DISCUSSION TITLE
Mix local sound-files into Jamulus session

### DISCUSSION THREAD

**Discussion by wperlich** [jamulussoftware/jamulus#1242] [Ideas]
We started to use Jamulus with our choir in Berlin. Jamulus is great as you can here the other voices sing. We are using our own private server for latency reasons.
We have prepared backups of the orchestra that support our singing. Is it possible to mix this track into Jamulus? If so, how to do this. If not, would be a good ides to realise this feature.

Wolfgang

**Comment by gene96817** [comment]
The last two choirs I introduced to Jamulus had a similar request. Since they are not an a cappella chorus, they needed a rehearsal pianist or an orchestral recording. While there are many ways to do this, perhaps we need an easy standard method for non-technical musicians. We probably should assume this is under the control of the director (who is not the server operator).

**Comment by ann0see** [comment]
Yes, this is possible of course. I did it on Debian 10/JACK + Ardour and it's very simimple. On Windows it involves a bit of work: Use Voicemeeter Banana and route the signal of your media player to Jamulus.

**Reply by gene96817** [reply]
Thinking about the Settings window, from Misc/Audio Channels, could there be a new option to include a second input channel? This could be in another place. Would we have to add another tool like Blackhole (which I still have not succeeded in using)?   We need to go from what is technically possible to what is possible for the ensemble's director.

**Reply by ann0see** [reply]
I think it's not that _easy_ to include an option like that. I know people would like to have this feature, but please keep https://github.com/jamulussoftware/jamulus/discussions/915 in mind. Also read hoffie's answer

**Comment by hoffie** [comment]
I'm using `mplayer -ao jack foo.mp3` and `qjackctl` to ensure proper routing. I'm also doing this with an additional Jamulus client directly next to the Jamulus server.

So, it's possible, although usability could be a bit better. ;)

I think something like a special client could be implemented (outside of Jamulus) to make such things easier similar to what I've proposed here with regards to a metronome: https://github.com/jamulussoftware/jamulus/discussions/1064#discussioncomment-411007

This hasn't gotten out of the ideas stage though. :(

**Reply by gene96817** [reply]
I just reread #1064. We could consider both discussions the same use case. I suggest we need a purely client-side solution (with no dependency on the user's relationship with the server). I also suggest any solution looking like a command line, while technically sufficient, is a non-starter for an app-only user.
